### PR TITLE
README and Documentation Updates

### PR DESCRIPTION
**Pull request by VVoruganti**
Fixes dev-428

- Updates the main README to use `uv` for package management commands and up to date environment variables
- Adds documentation for the Next.js application in the README in `www/README.md` 
- Creates a migration to setup local instance of supabase for local development
- Adds up to date prettier and eslint integrations for standard styling
- Updated `agent/agent/chain.py` to include `call` methods to make the discord bot work again without mirascope

Generally quality of life and documentation updates. Not much in here for the actual functionality of the project.

### CODE REVIEW

File: www/components/auth/signUp.tsx [hunk 1]
```diff
@@ -1,87 +1,93 @@
...
+          confirmButtonText: 'Close',
+          confirmButtonColor: '#3085d6',
+        });
+        return;
+      }
+      if (password.length < 6) {
+        await Swal.fire({
+          title: 'Insufficient Password',
+          icon: 'error',
+          text: 'Make sure the password is atleast 6 characters long',
```
Comment (bLopata) [inline www/components/auth/signUp.tsx:48]: Missing space `at least`

File: www/components/auth/signUp.tsx [hunk 2]
```diff
@@ -1,87 +1,93 @@
...
+        });
+        return;
+      }
+
+      const error = await handler(formData);
+      if (error) {
+        Swal.fire({
+          title: 'Something went wrong',
+          icon: 'error',
+          text: 'Please try again and make sure the password is atleast 6 characters long',
```
Comment (bLopata) [inline www/components/auth/signUp.tsx:60]: Missing space `at least`

File: www/utils/supabase/admin.ts
Comment (bLopata) [inline www/utils/supabase/admin.ts:1]: Any reason why this file is using double quotes instead of single quotes?

**Review by bLopata — COMMENTED** [review]
Couple of small nit/typo things. I want to pull it down locally and run the db migration and linter before I approve.

File: www/README.md [hunk 3]
```diff
@@ -1,48 +1,95 @@
...
 
-The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+## Supabase
 
-Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+### Supabase
 
-## Supabase Setup
+This project uses supabase for managing authentication and keeping track of
+stripe subscriptions.We recommend for testing and local development to use a
```
Comment (bLopata) [inline www/README.md:70]:
```suggestion
stripe subscriptions. We recommend for testing and local development to use a
```

File: www/README.md [hunk 4]
```diff
@@ -1,48 +1,95 @@
...
 
-## Deploy on Vercel
+- `NEXT_PUBLIC_POSTHOG_KEY` — The Posthog project key
+- `NEXT_PUBLIC_POSTHOG_HOST` — The Posthog host
 
-The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+## Supabase
 
-Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+### Supabase
```
Comment (bLopata) [inline www/README.md:67]:
```suggestion
### Setup
```

File: www/utils/supabase/admin.ts
Comment (VVoruganti) [reply www/utils/supabase/admin.ts:1]: Good catch, added the `utils/` folder to the `prettier` and `eslint` commands

**Review by bLopata — APPROVED** [review]
LGTM!

**Pull request merged**